.github/workflows/publish-arc.yaml

```diff
@@ -66,4 +58,6 @@ jobs:
             ${{ env.DOCKERHUB_USERNAME }}/actions-runner-controller:latest
             ${{ env.DOCKERHUB_USERNAME }}/actions-runner-controller:${{ env.VERSION }}
             ${{ env.DOCKERHUB_USERNAME }}/actions-runner-controller:${{ env.VERSION }}-${{ steps.vars.outputs.sha_short }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```

Comment:
You're using the same scope in this workflow and the canary one and therefore will overlap with each other. You should define a scope:

```yaml
cache-from: type=gha,scope=arc
cache-to: type=gha,mode=max,scope=arc
```